src/typechecker.rs

```diff
@@ -3449,6 +3449,20 @@ fn typecheck_function(
         resolve_type_var(function_return_type_id, parent_scope_id, project)
     };
 
+    // After infering the type we need to check again for main since it can only return c_int
```

Comment:
Spelling error:

inferring

:^)

Reply:
Thank you for pointing that out! I fixed it now.